articles/speakeasy-two-factor-authentication-in-nodejs/index.md

```diff
@@ -0,0 +1,245 @@
+
+### Goal
+In this tutorial, we will do a backend implementation of two-factor authentication using a token generated by the google authenticator app.
+
+### Table of contents
+- [Introduction](#introduction)
+- [Setting up the Node application](#setting-up-the-nodejs-application)
+- [Installing the dependencies](#installing-the-dependencies)
+- [Setting up the application](#setting-up-the-application)
+- [The node json databse](#the-node-json-databse)
+- [Sending requests using postman](#sending-requests-using-postman)
+- [Registering users](#registering-users)
+- [Verifying the user](#verifying-the-user)
+  - [Retriving user-id and temp secret from the database](#retrieving-id-and-temp-secret-from-the-database)
+  - [Generating verification token](#generating-verification-token)
+  - [Sending post request for verification response via postman](#sending-post-request-for-verification-response-via-postman)
+- [Validate the token ](#validate-the-token )
+- [Running the server](#running-the-server)
+- [Conclusion](#conclusion)
+- [Further Reading](#further-reading)
+### Introduction
+The Speakeasy library provides two-factor authentication using a one-time pin(OTP). The library provides an additional layer of security to an application's standards authentication process. Using the OTP, Speakeasy provides extra data required for account access.
```

Comment:
Add this section before `###Goal`.

Reply:
Firstly, I would suggest you to tell how authentication happens? Then, give a little intro about two-factor authentication. Later, you could explain more about Speakeasy library.

Also, add a link to the repo or the npm package of this library